server/authorization.go

```diff
@@ -46,14 +47,28 @@ type AuthorizationServer struct {
 
 // Register registers
 func (a *AuthorizationServer) Register(s *grpc.Server, handler *Handler) {
-	auth.RegisterAuthorizationServer(s, a)
+	envoy_auth.RegisterAuthorizationServer(s, a)
 	a.handler = handler
 }
 
 // Check does check
-func (a *AuthorizationServer) Check(ctx context.Context, req *auth.CheckRequest) (*auth.CheckResponse, error) {
+func (a *AuthorizationServer) Check(ctx gocontext.Context, req *envoy_auth.CheckRequest) (*envoy_auth.CheckResponse, error) {
+
+	var rootContext context.Context = a.handler
+	if a.handler.isMultitenant {
+		if env, override := req.Attributes.ContextExtensions[envContextKey]; override {
+			rootContext = &multitenantContext{
+				a.handler,
+				env,
+			}
+		} else {
+			err := fmt.Errorf("Envoy must be configured to send Apigee env in ContextExtensions[%s] in multitenant mode.", envContextKey)
+			log.Errorf(err.Error())
+			return nil, err
```

Comment:
What would Envoy do upon receiving a nil response? Would it make sense to send a denial?